src/Components/AdvancedSearch/Internal/QueryBuilder/QueryBuilderRow.tsx

```diff
@@ -0,0 +1,310 @@
+                        });
+                    });
+                }
+            );
+            setPropertyOptions(propertyComboboxOptions);
+        }
+    }, [isLoading, flattenedModelProperties]);
+
+    // Update operator options
+    useEffect(() => {
+        if (selectedProperty) {
+            const operatorOptions = getOperators(selectedProperty.data.type);
+            setOperatorOptions(operatorOptions);
+            if (operatorOptions.length) {
+                setSelectedOperator(operatorOptions[0]);
+            }
+        }
+    }, [selectedProperty]);
+
+    // Update snippet on any input change
+    useEffect(() => {
+        if (selectedProperty) {
+            updateSnippet(rowId, {
+                combinator: selectedCombinator
+                    ? selectedCombinator
+                    : getDefaultCombinator(),
+                operatorData: selectedOperator
+                    ? selectedOperator.data
+                    : getDefaultOperator(),
+                property: selectedProperty.data.name,
+                value: selectedValue
+                    ? selectedValue
+                    : getDefaultPropertyValues(selectedProperty.data.type)
+            });
+        }
+    }, [selectedCombinator, selectedOperator, selectedProperty, selectedValue]);
+
+    const onChangeOperator = (
+        _event: React.FormEvent<HTMLDivElement>,
+        option?: IDropdownOption<OperatorData>,
+        _index?: number
+    ) => {
+        setSelectedOperator(option);
+    };
+
+    const onChangeCombinator = (
+        _event: React.FormEvent<HTMLDivElement>,
+        option?: IDropdownOption,
+        _index?: number
+    ) => {
+        setSelectedCombinator(option.text);
+    };
+
+    const onChangePropertySelected = (
+        newValue: PropertyOption,
+        _actionMeta: SelectOptionActionMeta<PropertyOption>
+    ) => {
+        setSelectedProperty(newValue);
+        onChangeProperty(rowId, newValue.data?.name, newValue.data?.type);
+    };
+
+    const onChangeValueField = (
+        _event: React.SyntheticEvent<HTMLElement, Event>,
+        newValue?: string
+    ) => {
+        setSelectedValue(newValue);
+        onChangeValue(rowId, newValue);
+    };
+
+    const onChangeDropdownValue = (
+        _event: React.FormEvent<HTMLDivElement>,
+        option?: IDropdownOption<any>,
+        _index?: number
+    ) => {
+        setSelectedValue(option.text);
+        onChangeValue(rowId, option.text);
+    };
+
+    const propertySelectorStyles = reactSelectStyles(isRemoveDisabled);
+
+    const Group = (props) => (
+        <div>
+            <components.Group {...props} />
+        </div>
+    );
+
+    const Menu = (props) => (
+        <Callout
+            target={`#${propertySelectorId}`}
+            styles={classNames.subComponentStyles.propertyCallout}
+            isBeakVisible={false}
+        >
+            <BaseComponent theme={theme}>
+                <components.MenuList
```

Comment:
can we add a tooltip on each menu item hover? long values get cut off and you can't read them. 
![image](https://user-images.githubusercontent.com/57726991/187563285-abf0e448-d8a9-4801-a127-d59de0fb1323.png)

Reply:
react-select doesn't have this built-in I'll include this functionality later if there is feedback that we need it.